L1Trigger/L1TTrackMatch/plugins/L1TruthTrackFastJetProducer.cc

```diff
@@ -0,0 +1,218 @@
+
+// MC
+#include "SimTracker/TrackTriggerAssociation/interface/TTTrackAssociationMap.h"
+#include "SimDataFormats/TrackingAnalysis/interface/TrackingParticle.h"
+
+// geometry
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+
+#include <fastjet/JetDefinition.hh>
+
+#include <string>
+#include "TMath.h"
+#include "TH1.h"
+
+using namespace l1t;
+using namespace edm;
+using namespace std;
+
+//////////////////////////////
+//                          //
+//     CLASS DEFINITION     //
+//                          //
+//////////////////////////////
+
+class L1TruthTrackFastJetProducer : public edm::stream::EDProducer<> {
+public:
+  typedef TTTrack<Ref_Phase2TrackerDigi_> L1TTTrackType;
+  typedef std::vector<L1TTTrackType> L1TTTrackCollectionType;
+
+  explicit L1TruthTrackFastJetProducer(const edm::ParameterSet&);
+  ~L1TruthTrackFastJetProducer() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void produce(edm::Event&, const edm::EventSetup&) override;
+
+  // track selection criteria
+  const float trkZMax_;      // in [cm]
+  const float trkPtMin_;     // in [GeV]
+  const float trkEtaMax_;    // in [rad]
+  const int trkNStubMin_;    // minimum number of stubs
+  const int trkNPSStubMin_;  // minimum number of PS stubs
+  const double coneSize_;    // Use anti-kt with this cone size
+  const bool displaced_;     //use prompt/displaced tracks
+
+  const edm::EDGetTokenT<std::vector<TTTrack<Ref_Phase2TrackerDigi_> > > trackToken_;
+  edm::ESGetToken<TrackerTopology, TrackerTopologyRcd> tTopoToken_;
+  const edm::EDGetTokenT<TTTrackAssociationMap<Ref_Phase2TrackerDigi_> > ttTrackMCTruthToken_;
+};
+
+// constructor
+L1TruthTrackFastJetProducer::L1TruthTrackFastJetProducer(const edm::ParameterSet& iConfig)
+    : trkZMax_((float)iConfig.getParameter<double>("trk_zMax")),
+      trkPtMin_((float)iConfig.getParameter<double>("trk_ptMin")),
+      trkEtaMax_((float)iConfig.getParameter<double>("trk_etaMax")),
+      trkNStubMin_((int)iConfig.getParameter<int>("trk_nStubMin")),
+      trkNPSStubMin_((int)iConfig.getParameter<int>("trk_nPSStubMin")),
+      coneSize_((float)iConfig.getParameter<double>("coneSize")),
+      displaced_(iConfig.getParameter<bool>("displaced")),
+      trackToken_(consumes<std::vector<TTTrack<Ref_Phase2TrackerDigi_> > >(
+          iConfig.getParameter<edm::InputTag>("L1TrackInputTag"))),
+      tTopoToken_(esConsumes<TrackerTopology, TrackerTopologyRcd>(edm::ESInputTag("", ""))),
+      ttTrackMCTruthToken_(consumes<TTTrackAssociationMap<Ref_Phase2TrackerDigi_> >(
+          iConfig.getParameter<edm::InputTag>("MCTruthTrackInputTag"))) {
+  if (displaced_)
+    produces<TkJetCollection>("L1TruthTrackFastJetsExtended");
+  else
+    produces<TkJetCollection>("L1TruthTrackFastJets");
+}
+
+// destructor
+L1TruthTrackFastJetProducer::~L1TruthTrackFastJetProducer() {}
```

Comment:
Please remove this empty function and set the destructor to default in the class declaration.